src/ResetMap.jl

```diff
@@ -0,0 +1,169 @@
+import Base: isempty
+
+export ResetMap,
+       apply,
+       apply!
+
+"""
+    ResetMap{N<:Real} <: LazySet{N}
+
+Type that represents a lazy reset map.
+A reset map is a special case of an affine map ``Ax + b`` where the linear map
+``A`` is the identity matrix with zero entries in all reset dimensions, and the
+translation ``b`` is zero in all non-reset dimensions.
+
+### Fields
+
+- `X`      -- convex set
+- `resets` -- resets (a mapping from an index to a new value)
+
+### Example
+
+```jldoctest resetmap
+julia> X = BallInf([2.0, 2.0, 2.0], 1.0);
+
+julia> r = Dict(1 => 4.0, 3 => 0.0);
+
+julia> rm = ResetMap(X, r);
+
+```
+
+Here `rm` modifies the set `X` such that `x1` is reset to 1 and `x3` is reset to
+0, while `x2` is not modified.
+The corresponding affine map ``A x + b`` would be
+
+```math
+\begin{pmatrix} 0 & 0 & 0 \\ 0 & 1 & 0 \\ 0 & 0 & 0 \end{pmatrix} x +
+\begin{pmatrix} 4 & 0 & 0 \end{pmatrix}
+```
+
+Applying the reset map to a vector `x = [1.0, 2.0, 3.0]`, we get:
+
+```jldoctest resetmap
+julia> apply(rm, [1.0, 2.0, 3.0])
+3-element Array{Float64,1}:
+ 4.0
+ 2.0
+ 0.0
+```
+"""
+struct ResetMap{N<:Real} <: LazySet{N}
+    X::LazySet{N}
+    resets::Dict{Int, N}
+end
+
+"""
+    apply(rm::ResetMap{N}, x::AbstractVector{N}) where {N<:Real}
+
+Apply a reset map to a given vector.
+
+### Input
+
+- `rm` -- reset map
+- `x`  -- vector
+
+### Output
+
+A fresh vector corresponding to `x` after `rm` was applied.
+"""
+function apply(rm::ResetMap{N}, x::AbstractVector{N}) where {N<:Real}
+    return apply!(rm, copy(x))
+end
+
+"""
+    apply!(rm::ResetMap{N}, x::AbstractVector{N}) where {N<:Real}
+
+Apply a reset map to a given vector.
+
+### Input
+
+- `rm` -- reset map
+- `x`  -- vector (modified in this function)
+
+### Output
+
+The same (but see the Notes below) vector `x` but after `rm` was applied.
+
+### Notes
+
+The vector `x` is modified in-place if it has type `Vector` or `SparseVector`.
+Otherwise, we first create a new `Vector` from it.
+"""
+function apply!(rm::ResetMap{N}, x::AbstractVector{N}) where {N<:Real}
+    return substitute!(rm.resets, x)
+end
+
+
+# --- LazySet interface functions ---
+
+
+"""
+    dim(rm::ResetMap)
+
+Return the dimension of a reset map.
+
+### Input
+
+- `rm` -- reset map
+
+### Output
+
+The dimension of a reset map.
+"""
+function dim(rm::ResetMap)::Int
+    return dim(rm.X)
+end
+
+"""
+    σ(d::AbstractVector{N}, rm::ResetMap{N}) where {N<:Real}
+
+Return the support vector of a reset map.
+
+### Input
+
+- `d`  -- direction
+- `rm` -- reset map
+
+### Output
+
+The support vector in the given direction.
+If the direction has norm zero, the result depends on the wrapped set.
+"""
+function σ(d::AbstractVector{N}, rm::ResetMap{N}) where {N<:Real}
+    return apply!(rm, σ(d, rm.X))
```

Comment:
Is this correct? This formula coded is `A * σ(d, X) + b`. But if i start from the set, `A*X + b`, then: `σ(d, A*X + b) = Aσ(A^T*d) + b`, see [Prop. 2.2 here](https://juliareach.github.io/LazySets.jl/latest/man/polyhedral_approximations.html#Support-Vector-1).

Reply:
The outer `A` just makes some entries 0, namely those that are reset in `b`. The inner `A^T = A`  makes the same entries in `d` 0. I thought that the 0-entries do not hurt, but now I am not sure anymore.
My new formula derived from yours would be `apply!(rm, σ(d*, rm.X))`, where `d*` has all entries reset to zero that are reset in `rm`.

Reply:
:100: Your hunch was right: If the support vector of `X` is unique, it should work, but if there are choices, then the result can be wrong. Consider a `Ball1` in 2D, the direction `d = [1.0, 1.0]`, and the reset `2 => 0.0`. If `σ(d, X)` returns `[1.0, 0.0]`, then the result will be `[1.0, 0.0]` (correct), but if `σ(d, X)` returns `[0.0, 1.0]`, then the result will be `[0.0, 0.0]`.
Great that you found this. It would have been a painful debugging at some point.

Reply:
Saw the new function, looks good 👍